src/test/java/org/opensearch/knn/integ/BinaryIndexIT.java

```diff
@@ -0,0 +1,242 @@
+        // Create Index
+        String indexName = getCurrentMethodName().toLowerCase(Locale.ROOT);
+        String fieldName = "my-vector";
+        Exception e = expectThrows(
+            ResponseException.class,
+            () -> createKnnHnswBinaryIndex(true, KNNEngine.NMSLIB, indexName, fieldName, 16, null)
+        );
+        assertTrue(e.getMessage(), e.getMessage().contains("only supported for [faiss] engine"));
+    }
+
+    @SneakyThrows
+    public void testFaissHnswBinary_whenEncodedWithSQ_thenThrowException() {
+        // Create Index
+        String indexName = getCurrentMethodName().toLowerCase(Locale.ROOT);
+        String fieldName = "my-vector";
+        Exception e = expectThrows(
+            Exception.class,
+            () -> createKnnHnswBinaryIndex(true, KNNEngine.FAISS, indexName, fieldName, 16, ENCODER_SQ)
+        );
+        assertTrue(e.getMessage(), e.getMessage().contains("does not support sq encoder"));
+    }
+
+    @SneakyThrows
+    public void testFaissHnswBinary_whenRadialSearch_thenThrowException() {
+        // Create Index
+        String indexName = getCurrentMethodName().toLowerCase(Locale.ROOT);
+        String fieldName = "my-vector";
+        createKnnHnswBinaryIndex(true, KNNEngine.FAISS, indexName, fieldName, 16, null);
+
+        // Query
+        float[] queryVector = { (byte) 0b10001111, (byte) 0b10000000 };
+        Exception e = expectThrows(Exception.class, () -> runRnnQuery(indexName, fieldName, queryVector, 1, 4));
+        assertTrue(e.getMessage(), e.getMessage().contains("Binary data type does not support radial search"));
+    }
+
+    private float getRecall(final Set<String> truth, final Set<String> result) {
+        // Count the number of relevant documents retrieved
+        result.retainAll(truth);
+        int relevantRetrieved = result.size();
+
+        // Total number of relevant documents
+        int totalRelevant = truth.size();
+
+        // Calculate recall
+        return (float) relevantRetrieved / totalRelevant;
+    }
+
+    private List<KNNResult> runRnnQuery(final String indexName, final String fieldName, final float[] queryVector, final int minScore, final int size)
+        throws Exception {
+        XContentBuilder query = XContentFactory.jsonBuilder()
+            .startObject().startObject(QUERY).startObject(KNN).startObject(fieldName)
+            .field(VECTOR, queryVector)
+            .field(MIN_SCORE, minScore)
+            .endObject().endObject().endObject().endObject();
+        Response response = searchKNNIndex(indexName, query, size);
+        return parseSearchResponse(EntityUtils.toString(response.getEntity()), fieldName);
+    }
+
+    private List<KNNResult> runKnnQuery(final String indexName, final String fieldName, final float[] queryVector, final int k)
+        throws Exception {
+        Response response = searchKNNIndex(indexName, KNNQueryBuilder.builder().fieldName(fieldName).vector(queryVector).k(k).build(), k);
```

Comment:
Better to use `searchKNNIndex(String index, XContentBuilder xContentBuilder, int resultSize)` to mock the request from end

Reply:
What do you mean with `mock the request from end`?

Reply:
I mean using XContentBuilder can better mock request from end user api call than use KNNQueryBuilder

Reply:
Got it. Fait enough.